### PR TITLE
Support for third party engines

### PR DESCRIPTION
**Pull request by jpechane**

### CODE REVIEW

File: mysql/MySqlParser.g4 [hunk 1]
```diff
@@ -1889,9 +1889,13 @@ engineName
     : ARCHIVE | BLACKHOLE | CSV | FEDERATED | INNODB | MEMORY
     | MRG_MYISAM | MYISAM | NDB | NDBCLUSTER | PERFORMANCE_SCHEMA
     | TOKUDB
+    | customEngineName
```
Comment (KvanTTT) [inline mysql/MySqlParser.g4]: Why don't use just an `ID` token without adding new `customEngineName` rule? Also, you can use an element label in this case: `customEngineName=ID`.

Comment (jpechane) [reply mysql/MySqlParser.g4]: @KvanTTT I did it intentionally, so the generated code will contain reference to not generic id. The label is probably a better solution so I'll try it out and modify the PR if it works as expected

Comment (jpechane) [reply mysql/MySqlParser.g4]: @KvanTTT Updated

**Comment by KvanTTT** [comment]
Thanks

**Pull request merged**